src/vmspawn/vmspawn-util.c

```diff
@@ -44,6 +44,21 @@ int qemu_check_kvm_support(void) {
         return -errno;
 }
 
+int qemu_check_vsock_support(void) {
```

Comment:
hmm so /dev/vhost-vsock is one of those device nodes that are created via tmpfiles even without the module backing it being loaded. This means this check will succeed even if the backing module is missing.

I think it would make sense to try to open the device, and just close it immediately again if that worked. That will trigger the kmod logic, and will result in ENODEV or so if the functionality is not actually available.

(please add a comment explaining this)

Reply:
Fixed in b0f65acd48.